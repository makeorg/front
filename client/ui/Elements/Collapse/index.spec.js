import React from 'react';
import renderer from 'react-test-renderer';
import snapshotDiff from 'snapshot-diff';
import { Collapse } from './index';

jest.mock('Client/ui/Elements/CollapseElements', () => ({
  TileWithCollapseWrapperStyle: 'TileWithCollapseWrapperStyle',
  CollapseWrapperStyle: 'TileSeparatorStyle',
  CollapseTriggerStyle: 'CollapseTriggerStyle',
  CollapseIconStyle: 'CollapseIconStyle',
  CollapseContentStyle: 'CollapseContentStyle',
  CollapseSeparatorStyle: 'CollapseSeparatorStyle',
}));

jest.mock('Client/ui/Svg/elements', () => ({
  SvgAngleArrowRight: 'SvgAngleArrowRight',
}));

describe('Collapse', () => {
  const title = 'foo';
  const children = 'bar';

  it('must match the snapshot with default Props', () => {
    const component = renderer
      .create(<Collapse title={title}>{children}</Collapse>)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('must match the diff snapshot between Expand vs Collapse Props', () => {
    const expandedComponent = renderer
      .create(
        <Collapse title={title} open>
          {children}
        </Collapse>
      )
      .toJSON();
    const defaultCollapse = renderer
      .create(<Collapse title={title}>{children}</Collapse>)
      .toJSON();
    expect(snapshotDiff(expandedComponent, defaultCollapse)).toMatchSnapshot();
  });

  it('must match the diff snapshot between Tile vs default Collapse Style', () => {
    const collapseWithTile = renderer
      .create(
        <Collapse title={title} withTileStyle>
          {children}
        </Collapse>
      )
      .toJSON();
    const defaultCollapse = renderer
      .create(<Collapse title={title}>{children}</Collapse>)
      .toJSON();
    expect(snapshotDiff(collapseWithTile, defaultCollapse)).toMatchSnapshot();
  });
});
