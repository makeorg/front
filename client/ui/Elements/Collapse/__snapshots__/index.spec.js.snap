// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse must match the diff snapshot between Expand vs Collapse Props 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`Collapse must match the diff snapshot between forceExpandOnDesktop vs default Props 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`Collapse must match the snapshot with default Props 1`] = `
<TileSeparatorStyle
  aria-expanded={false}
>
  <CollapseTriggerStyle
    aria-disabled={false}
    aria-label="common.expand_collapse"
    disabled={false}
    onClick={[Function]}
  >
    <ThirdLevelTitleStyle>
      foo
    </ThirdLevelTitleStyle>
    <CollapseIconStyle
      aria-hidden={true}
      iscollapsed={true}
    >
      <SvgAngleArrowRight />
    </CollapseIconStyle>
  </CollapseTriggerStyle>
  <CollapseContentStyle
    aria-hidden={true}
    forcedexpand={false}
    iscollapsed={true}
  >
    <CollapseSeparatorStyle
      aria-hidden={true}
    />
    bar
  </CollapseContentStyle>
</TileSeparatorStyle>
`;
